source/dubregistry/web.d

```diff
@@ -305,6 +305,18 @@ class DubRegistryWebFrontend {
 				return url;
 			}
 
+			bool hasManagementAccess;
+			bool isMine;
+			if (req.session) {
+				auto userId = User.ID.fromString(req.session.get!string("userID"));
+				hasManagementAccess = packinfo.hasPermissions(
```

Comment:
I don't know much about the context of those files but this line looks a little strange to me.

Reply:
Btw, other than that, I didn't noticed anything strange in the code.

Reply:
this is used to show the "Edit this package" button on the view package page - since we potentially don't have a user, we need to fetch it from the session here. The userId / session reading code is just moved from inside the .dt template into the .d file here. The first hasPermissions check checks if we should show the button, the second checks if the package is owned by the logged in user (to show a different text below the button)

Reply:
Oh yea, what I found strange was using the `readonly` permission for hasManagementAccess, it looked very strange.